core/server/src/main/java/alluxio/master/AlluxioMasterRestServiceHandler.java

```diff
@@ -0,0 +1,323 @@
+
+    // Get all counters.
+    Map<String, Counter> counters = metricRegistry.getCounters();
+
+    // Only the gauge for pinned files is retrieved here, other gauges are statistics of free/used
+    // spaces, those statistics can be gotten via other REST apis.
+    String filesPinnedProperty = CommonUtils.argsToString(".",
+        MasterContext.getMasterSource().getName(), MasterSource.FILES_PINNED);
+    Gauge filesPinned = metricRegistry.getGauges().get(filesPinnedProperty);
+
+    // Get values of the counters and gauges and put them into a metrics map.
+    SortedMap<String, Long> metrics = new TreeMap<>();
+    for (Map.Entry<String, Counter> counter : counters.entrySet()) {
+      metrics.put(counter.getKey(), counter.getValue().getCount());
+    }
+    metrics.put(filesPinnedProperty, ((Integer) filesPinned.getValue()).longValue());
+
+    return Response.ok(metrics).build();
+  }
+
+  /**
+   * @summary get the start time of the master
+   * @return the response object
+   */
+  @GET
+  @Path(GET_START_TIME_MS)
+  @ReturnType("java.lang.Long")
+  public Response getStartTimeMs() {
+    return Response.ok(mMaster.getStartTimeMs()).build();
+  }
+
+  /**
+   * @summary get the uptime of the master
+   * @return the response object
+   */
+  @GET
+  @Path(GET_UPTIME_MS)
+  @ReturnType("java.lang.Long")
+  public Response getUptimeMs() {
+    return Response.ok(mMaster.getUptimeMs()).build();
+  }
+
+  /**
+   * @summary get the version of the master
+   * @return the response object
+   */
+  @GET
+  @Path(GET_VERSION)
+  @ReturnType("java.lang.String")
+  public Response getVersion() {
+    return Response.ok(Version.VERSION).build();
+  }
+
+  /**
+   * @summary get the total capacity of all workers in bytes
+   * @return the response object
+   */
+  @GET
+  @Path(GET_CAPACITY_BYTES)
+  @ReturnType("java.lang.Long")
+  public Response getCapacityBytes() {
+    return Response.ok(mBlockMaster.getCapacityBytes()).build();
+  }
+
+  /**
+   * @summary get the used capacity
+   * @return the response object
+   */
+  @GET
+  @Path(GET_USED_BYTES)
+  @ReturnType("java.lang.Long")
+  public Response getUsedBytes() {
+    return Response.ok(mBlockMaster.getUsedBytes()).build();
+  }
+
+  /**
+   * @summary get the free capacity
+   * @return the response object
+   */
+  @GET
+  @Path(GET_FREE_BYTES)
+  @ReturnType("java.lang.Long")
+  public Response getFreeBytes() {
+    return Response.ok(mBlockMaster.getCapacityBytes() - mBlockMaster.getUsedBytes()).build();
+  }
+
+  /**
+   * @summary get the total ufs capacity in bytes, a negative value means the capacity is unknown.
+   * @return the response object
+   */
+  @GET
+  @Path(GET_UFS_CAPACITY_BYTES)
+  @ReturnType("java.lang.Long")
+  public Response getUfsCapacityBytes() {
+    try {
+      return Response.ok(mUfs.getSpace(mUfsRoot, UnderFileSystem.SpaceType.SPACE_TOTAL)).build();
+    } catch (IOException e) {
+      LOG.warn(e.getMessage());
```

Comment:
if we always use warn in rest handlers, then I am fine---just check if we are consistent.